### PR TITLE
Refactor dns flows and workers

### LINKED ISSUE #28: Filter bad words in domain names
This is a child issues of #8 to minimize the original scope of work.

- Filter inappropriate domain name

### LINKED ISSUE #110: Enforce a configurable limit on the number of DNS records per user
We need to be able to set an environment variable that imposes a limit on the number of DNS records a user can have.  When the user tries to add a new record, their current record count should be compared against this limit, and the operation should fail if it will mean exceeding the limit.

I'm not sure what the right way to enforce this is (e.g., in the db, in the model code, etc).

### LINKED ISSUE #315: Break DNS update into two phases
Currently all of the steps to add/update record immediately go into queue which gives frontend long time to see the response. 

Instead the system updates DB immediately without waiting in a queue whole time. Then, queue will be run to update Route 53, poll the status, and update the Route 53 status in DB. 

This will give much better experience to the user.

### LINKED ISSUE #333: DNS flow refactors
I'd like to start a discussion about this file's TS declarations, because I'm quite certain it's wrong:
https://github.com/DevelopingSpace/starchart/blob/main/app/queues/dns/dns-flow.server.ts
First, we are trying to use `JobRecord`  for absolutely everything. Basically all the workers and exported flow initiator functions all use this, and we are trying to twist it to suite the specific use-case. This causes some very hard to follow code, and inappropriate use.
I.e.,  Line 17, `name`  refers to a subdomain only (which was a breaking change, but let's not get bogged down)
In Line 18, using name (subdomain) we create `fullRecordName` which is an FQDN
In line 24, we pass this FQDN down as `name`,
so in line 17 `name` was a subdomain, in line 24 `name` is FQDN and they are both using the same TS interface. This is wrong.

In my humble opinion:
- Each worker should be it's own file and should export an interface, of what it exactly expects as input params (`data`)
- [dns-flow.server.ts](https://github.com/DevelopingSpace/starchart/blob/main/app/queues/dns/dns-flow.server.ts) should import those interfaces and use the correct interface for each `data`  attribute in the flow generators
- There also should be a separate TS interface/type for each flow initiator. It can reuse the worker's interface, but should be specific.
- We should start using `subdomain`  and `fqdn`  atribute and variable names to signify what we are storing in it, `name` is not specific enough (see above issue)
- We need a way to pass down either FQDN or subdomain to the appropriate flow initiators, flow initiators should generate the fqdn if subdomain was provided. This is needed for the ACME integration (and probably other later code as well)
- Input validation should be part of the flow initiator, not the worker. This way we can synchronously return an error to the caller, and it's not the subsequent async job that's going to fail

### PR DESCRIPTION
**Pull request by Genne23v**
Close #315 
Close #333 

- Split DNS module into two phases (DB and DNS)
- Specify type to each flow, worker and job
- Change name to subdomain or fqdn to distinguish 

### How to test
Currently integration is only set up in `/dev` route. You don't need any environment setup to test. 

I'm not sure how DNS handling should be initiated after DB update (same frontend call or background job). 
@dadolhay This update requires DB table ID in `addDnsRequest` call in your function. Please let me know if you have any concern with this approach.

### CODE REVIEW

**Comment by humphd** [comment]
@Genne23v it would have been better to separate the new functionality you've added from the refactoring.  Is there any way to split this PR into smaller PRs?

If not, I'll review it as is.  But this isn't ideal.

**Comment by Genne23v** [comment]
I can separate. I will push an update soon.

**Comment by Genne23v** [comment]
@humphd  I have removed new functionalities although they are very small. I would appreciate your review.

**Comment by Genne23v** [comment]
@humphd @dadolhay I connected DNS module with frontend. It's functional, but it needs more backend error validation and display the error somewhere in frontend. Let me know if anything needs to be added.

File: app/routes/__index/domains/new.tsx [hunk 1]
```diff
@@ -45,16 +47,28 @@ export const action = async ({ request }: ActionArgs) => {
...
-  // Create a pending record... for now
-  // This will most likely be replaced by some other logic
-  await addDnsRequest({
-    username: user.username,
-    type: data.type,
-    name: data.name,
-    value: data.value,
-  });
+  try {
+    const recordId = await createUserRecord({
```
Comment (dadolhay) [inline app/routes/__index/domains/new.tsx:51]: This complexity should be abstracted away from the rest of the system. You could move this into the addDnsRequest flow creator (not in a worker, in the flow creator itself)

File: app/routes/dev.tsx [hunk 2]
```diff
@@ -12,8 +12,14 @@ import logger from '~/lib/logger.server';
 import { requireUser, requireUsername } from '~/session.server';
 import { addNotification } from '~/queues/notifications/notifications.server';
 import { addCertRequest } from '~/queues/certificate/certificate-flow.server';
+import { addDnsRequest } from '~/queues/dns/add-record-flow.server';
+import { updateDnsRequest } from '~/queues/dns/update-record-flow.server';
+import { deleteDnsRequest } from '~/queues/dns/delete-record-flow.server';
+import { createUserRecord, updateUserRecord } from '~/lib/record-db.server';
```
Comment (dadolhay) [inline app/routes/dev.tsx:18]: Both create and update flow creator should handle the required DB changes. This should be abstracted away.

File: app/lib/record-db.server.ts
[the remark this answers is not in the document]
Comment (Genne23v) [reply app/lib/record-db.server.ts:1]: I see the point. I think showing the right error message is more important than any minor benefit by having this file.

File: app/queues/dns/workers/poll-dns-status-worker.server.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+export const checkDnsStatusQueueName = 'check-dns-status';
+
+export const checkDnsStatusWorker = new Worker(
+  checkDnsStatusQueueName,
+  async (job) => {
+    logger.debug('Checking Route53 readiness');
+
+    const childJobValues: { [jobKey: string]: string } = await job.getChildrenValues();
+    const dnsRecordUpdateJobKey = Object.keys(childJobValues).find((item) =>
```
[the remark this answers is not in the document]
Comment (Genne23v) [reply app/queues/dns/workers/poll-dns-status-worker.server.ts:15]: Yeah, I saw the code. Thanks for reminding

File: app/queues/certificate/order-creator-worker.server.ts [hunk 4]
```diff
@@ -40,8 +40,9 @@ const handleChallenges = ({
     await addDnsRequest({
       username,
       type: 'TXT',
-      name: domain,
+      subdomain: domain,
```
[the remark this answers is not in the document]
Comment (Genne23v) [reply app/queues/certificate/order-creator-worker.server.ts:43]: I declared the type as following. 
```js
export type Name = { subdomain: string } | { fqdn: string };
interface BaseAddDnsRequestData {
  username: Record['username'];
  type: Record['type'];
  value: Record['value'];
}
export type AddDnsRequestData = BaseAddDnsRequestData & Name;
```
But the behaviour is strange. It accepts `subdomain` in frontend, but not accepting `subdomain` in backend. Am I using it wrong?

**Comment by Genne23v** [comment]
@dadolhay @humphd I updated the PR. I would appreciate another round of review!

File: app/queues/dns/add-record-flow.server.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+import { checkDnsStatusQueueName } from './workers/poll-dns-status-worker.server';
+import { syncDbStatusQueueName } from './workers/sync-db-status-worker.server';
+import { createUserRecord } from '~/models/record.server';
+
+import type { FlowJob } from 'bullmq';
+import type { Record } from '@prisma/client';
+import type { DnsUpdaterData } from './workers/dns-update-worker.server';
+import type { DbRecordSynchronizerData } from './workers/sync-db-status-worker.server';
+
+export type Name = { subdomain: string } | { fqdn: string };
```
Comment (humphd) [inline app/queues/dns/add-record-flow.server.ts:14]: `Name` is really generic, can we do better?  `DomainName`, or `DomainNameComponent` or something, which could be a `subdomain` or `fqdn`?  Doesn't have to be that, but `Name` is not going to work.  @dadolhay will have thoughts.

File: app/queues/dns/add-record-flow.server.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+const flowProducer = new FlowProducer({ connection: redis });
+
+export const addDnsRequest = async (data: AddDnsRequestData) => {
+  const { username, type, value } = data;
+
+  let subdomain: string;
+  if ('subdomain' in data) {
+    subdomain = data.subdomain;
+  } else {
+    throw new Error('fqdn is not expected');
```
Comment (humphd) [inline app/queues/dns/add-record-flow.server.ts:40]: You could use [`tiny-invariant`](https://www.npmjs.com/package/tiny-invariant) (already in our deps) here:

```ts
invariant('subdomain' in data, 'expected subdomain vs. fqdn');
const subdomain = data.subdomain;
```

File: app/queues/dns/delete-record-flow.server.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  value: Record['value'];
+}
+export type DeleteDnsRequestData = BaseDeleteDnsRequestData & Name;
+
+const flowProducer = new FlowProducer({ connection: redis });
+
+export const deleteDnsRequest = async (data: DeleteDnsRequestData) => {
+  const { id, username, type, value } = data;
+
+  let subdomain: string;
```
Comment (humphd) [inline app/queues/dns/delete-record-flow.server.ts:28]: Same deal here, use `tiny-invariant` to throw if your condition is not met.

File: app/queues/dns/delete-record-flow.server.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+import { syncDbStatusQueueName } from './workers/sync-db-status-worker.server';
+import { WorkType } from './add-record-flow.server';
+
+import type { FlowJob } from 'bullmq';
+import type { Record } from '@prisma/client';
+import type { DnsUpdaterData } from './workers/dns-update-worker.server';
+import type { DbRecordSynchronizerData } from './workers/sync-db-status-worker.server';
+import type { Name } from './add-record-flow.server';
+
+interface BaseDeleteDnsRequestData {
```
Comment (humphd) [inline app/queues/dns/delete-record-flow.server.ts:15]: Instead of needing a name like this, you can do it all at once:

```ts
export type DeleteDnsRequestData = {
  id: Record['id'];
  username: Record['username'];
  type: Record['type'];
  value: Record['value'];
} & Name;
```

File: app/queues/dns/update-record-flow.server.ts [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+interface BaseUpdateDnsRequestData {
+  id: Record['id'];
+  username: Record['username'];
+  type: Record['type'];
+  value: Record['value'];
+  description?: Record['description'];
+  course?: Record['course'];
+  ports?: Record['ports'];
+}
+export type UpdateDnsRequestData = BaseUpdateDnsRequestData & Name;
```
Comment (humphd) [inline app/queues/dns/update-record-flow.server.ts:26]: Same thing here, inline this type and get rid of the `Base*` naming

File: app/queues/dns/update-record-flow.server.ts [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+const flowProducer = new FlowProducer({ connection: redis });
+
+export const updateDnsRequest = async (data: UpdateDnsRequestData) => {
+  const { username, type, value, id } = data;
+
+  let subdomain: string;
```
Comment (humphd) [inline app/queues/dns/update-record-flow.server.ts:33]: tiny-invariant

File: app/queues/dns/workers/dns-update-worker.server.ts [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+export interface DnsUpdaterData {
+  workType: WorkType;
+  username: Record['username'];
+  type: Record['type'];
+  fqdn: string;
+  value: Record['value'];
+}
+
+export type dnsRecordUpdateJobResult = string;
```
Comment (humphd) [inline app/queues/dns/workers/dns-update-worker.server.ts:20]: I think this should start with a `D` vs. `d`

File: app/queues/dns/workers/poll-dns-status-worker.server.ts [hunk 12]
```diff
@@ -0,0 +1,39 @@
+import { Worker } from 'bullmq';
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import { getChangeStatus } from '~/lib/dns.server';
+import { dnsUpdateQueueName } from './dns-update-worker.server';
+import { getChildrenValuesOfQueueName } from '~/utils';
+
+export const checkDnsStatusQueueName = 'check-dns-status';
+
+// interface PollDnsStatus {
```
Comment (humphd) [inline app/queues/dns/workers/poll-dns-status-worker.server.ts:10]: Can we remove this?

File: app/queues/dns/workers/poll-dns-status-worker.server.ts [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    logger.debug('Checking Route53 readiness');
+
+    const childrenValues = await getChildrenValuesOfQueueName<string>({
+      queueName: dnsUpdateQueueName,
+      job,
+    });
+
+    const [changeId] = Object.values(childrenValues);
+
+    // if (!dnsUpdateJobKey) {
```
Comment (humphd) [inline app/queues/dns/workers/poll-dns-status-worker.server.ts:26]: Why is this here?

File: app/routes/__index/domains/new.tsx [hunk 14]
```diff
@@ -45,16 +46,20 @@ export const action = async ({ request }: ActionArgs) => {
...
+      type: data.type,
+      subdomain: data.name,
+      value: data.value,
+    });
 
-  return redirect(`/domains`);
+    return redirect(`/domains`);
+  } catch (error) {
+    logger.warn('Add DNS request error', error);
+    //Need to display an error response
```
Comment (humphd) [inline app/routes/__index/domains/new.tsx:60]: Please file that as an issue to fix later.

File: app/queues/dns/update-record-flow.server.ts [hunk 15]
```diff
@@ -0,0 +1,108 @@
+import { FlowProducer } from 'bullmq';
+import { redis } from '~/lib/redis.server';
+import { buildDomain } from '~/utils';
+import { dnsUpdateQueueName } from './workers/dns-update-worker.server';
+import { checkDnsStatusQueueName } from './workers/poll-dns-status-worker.server';
+import { syncDbStatusQueueName } from './workers/sync-db-status-worker.server';
+import { WorkType } from './add-record-flow.server';
+import type { RecordUpdate } from '~/models/record.server';
```
Comment (humphd) [inline app/queues/dns/update-record-flow.server.ts:8]: Move this down to the types imports below

File: app/queues/dns/update-record-flow.server.ts [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+    name: subdomain,
+    value: data.value,
+    username: data.username,
+    description: data.description,
+    course: data.course,
+    ports: data.ports,
+  };
+
+  // Before running workflow, update the record in DB
+  await updateRecordById(dbRecordUpdate);
```
Comment (humphd) [inline app/queues/dns/update-record-flow.server.ts:53]: If you avoid declaring the object before you use it, you don't need a type:

```ts
// {...} will get typed automatically, and you don't have to import/set the type
await updateRecordById({...});
```

File: app/models/record.server.ts [hunk 17]
```diff
@@ -1,59 +1,77 @@
 import { RecordStatus } from '@prisma/client';
-import { prisma } from '~/db.server';
 import dayjs from 'dayjs';
+import { prisma } from '~/db.server';
 
 import type { Record } from '@prisma/client';
 export type { Record } from '@prisma/client';
 
-export async function getRecordsByUsername(username: Record['username']) {
+export interface RecordCreation {
```
Comment (humphd) [inline app/models/record.server.ts:8]: These two types are clunky.  A type should describe a thing vs an action.

Instead of these names, can you:

1. Use inline types in this file, like `Pick<Record, 'username' | 'type' | 'name' | 'value'>`
2. Instead of using the exported type in other files, inline the objects you pass (I have a note below showing you what I mean)

**Review by humphd — CHANGES_REQUESTED** [review]
I've made an initial set of comments.  Looking good.

File: app/models/record.server.ts [hunk 18]
```diff
@@ -1,59 +1,77 @@
...
+  name: Record['name'];
+  value: Record['value'];
+}
+
+export interface RecordUpdate {
+  id: Record['id'];
+  type: Record['type'];
+  name: Record['name'];
+  value: Record['value'];
+  username?: Record['username'];
```
[the remark this answers is not in the document]
Comment (Genne23v) [reply app/models/record.server.ts:20]: This is Data type only for DB update. So the username is not mandatory. I hope it makes sense. If this is not aligned with frontend or something, I will update.

File: app/models/record.server.ts [hunk 19]
```diff
@@ -1,59 +1,77 @@
...
-      name,
-      value,
-      status,
-      username,
-      description,
-      course,
-      ports,
-      expiresAt,
-      lastNotified,
+      ...data,
```
[the remark this answers is not in the document]
Comment (Genne23v) [reply app/models/record.server.ts:67]: Thank you for the recommendation. Will be updated!

File: app/queues/dns/add-record-flow.server.ts (hunk 5 above)
Comment (Genne23v) [reply app/queues/dns/add-record-flow.server.ts:14]: It's been one of the most difficult tasks for me to name things related to domain. I would like to suggest `DomainType` as it's either partial domain of full domain.

File: app/queues/dns/delete-record-flow.server.ts (hunk 8 above)
Comment (Genne23v) [reply app/queues/dns/delete-record-flow.server.ts:15]: It's been so great for me to learn all these typing techniques. Thank you

File: app/queues/dns/workers/dns-update-worker.server.ts (hunk 11 above)
Comment (Genne23v) [reply app/queues/dns/workers/dns-update-worker.server.ts:20]: Will be fixed

File: app/queues/dns/workers/poll-dns-status-worker.server.ts (hunk 12 above)
Comment (Genne23v) [reply app/queues/dns/workers/poll-dns-status-worker.server.ts:10]: Sorry that I forgot removing the comments

File: app/routes/__index/domains/new.tsx (hunk 14 above)
Comment (Genne23v) [reply app/routes/__index/domains/new.tsx:60]: I wrote #364

File: app/models/record.server.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (humphd) [reply app/models/record.server.ts:20]: It's fine.  Why would you ever include it if it's not required?  I'd remove it if it's optional.

File: app/queues/dns/add-record-flow.server.ts (hunk 5 above)
Comment (humphd) [reply app/queues/dns/add-record-flow.server.ts:14]: > "There are only two hard things in Computer Science: cache invalidation and naming things." --Phil Karlton (Netscape)

You're a true programmer when you hit the hardest problems in CS. Welcome!

The more you can avoid needing to name things, the fewer names you have to come up with.

I don't think `DomainType` is wrong, but `DomainName` or even `Domain` are better imho, since both a `subdomain` and a `fully qualified domain name` are both `domains`.

Comment (dadolhay) [reply app/queues/dns/add-record-flow.server.ts:14]: I'd drobably name this `FqdnOrSubdomain`

Comment (dadolhay) [reply app/queues/dns/add-record-flow.server.ts:14]: and yes, separating stuff, race conditions, flaky tests, finding others` [monkey patches](https://en.wikipedia.org/wiki/Monkey_patch) and naming stuff are some of the hardest things in sw development

Comment (Genne23v) [reply app/queues/dns/add-record-flow.server.ts:14]: There could be many different opinions, but to me `FqdnOrSubdomain` makes sense a lot. Thanks for the suggestion.

File: app/queues/dns/add-record-flow.server.ts [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+import { syncDbStatusQueueName } from './workers/sync-db-status-worker.server';
+import { createUserRecord } from '~/models/record.server';
+
+import type { FlowJob } from 'bullmq';
+import type { Record } from '@prisma/client';
+import type { DnsUpdaterData } from './workers/dns-update-worker.server';
+import type { DbRecordSynchronizerData } from './workers/sync-db-status-worker.server';
+
+export type FqdnOrSubdomain = { subdomain: string } | { fqdn: string };
+interface BaseAddDnsRequestData {
```
Comment (dadolhay) [inline app/queues/dns/add-record-flow.server.ts:16]: Can we use pick here too? Genuine question, I really don't know because of the optional attributes

File: app/queues/dns/add-record-flow.server.ts [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+  update = 'update',
+  delete = 'delete',
+}
+
+const flowProducer = new FlowProducer({ connection: redis });
+
+export const addDnsRequest = async (data: AddDnsRequestData) => {
+  const { username, type, value } = data;
+
+  invariant('subdomain' in data, 'expected subdomain vs. fqdn');
```
Comment (dadolhay) [inline app/queues/dns/add-record-flow.server.ts:37]: If we are throwing an exception if fqdn is provided, then there is no point in accepting it in the first place.

If the whole system gravitates toward only working with a subdomain, then let's not accept fqdn here, I'll alter the dns worker send a subdomain instead

File: app/queues/dns/delete-record-flow.server.ts [hunk 22]
```diff
@@ -0,0 +1,86 @@
...
+  type: Record['type'];
+  value: Record['value'];
+} & FqdnOrSubdomain;
+
+const flowProducer = new FlowProducer({ connection: redis });
+
+export const deleteDnsRequest = async (data: DeleteDnsRequestData) => {
+  const { id, username, type, value } = data;
+
+  invariant('subdomain' in data, 'expected subdomain vs. fqdn');
```
Comment (dadolhay) [inline app/queues/dns/delete-record-flow.server.ts:28]: same as above

File: app/queues/dns/update-record-flow.server.ts [hunk 23]
```diff
@@ -0,0 +1,101 @@
...
+  course?: Record['course'];
+  ports?: Record['ports'];
+} & FqdnOrSubdomain;
+
+const flowProducer = new FlowProducer({ connection: redis });
+
+export const updateDnsRequest = async (data: UpdateDnsRequestData) => {
+  const { username, type, value, id } = data;
+
+  invariant('subdomain' in data, 'expected subdomain vs. fqdn');
```
Comment (dadolhay) [inline app/queues/dns/update-record-flow.server.ts:32]: same here

File: app/queues/dns/workers/dns-update-worker.server.ts [hunk 24]
```diff
@@ -0,0 +1,49 @@
...
+import type { WorkType } from '../add-record-flow.server';
+import type { Record } from '@prisma/client';
+
+export const dnsUpdateQueueName = 'dns-record-update';
+
+export interface DnsUpdaterData {
+  workType: WorkType;
+  username: Record['username'];
+  type: Record['type'];
+  fqdn: string;
```
Comment (dadolhay) [inline app/queues/dns/workers/dns-update-worker.server.ts:16]: If we rely on subdomain in the whole process, let's use that here too, and generate fqdn from the username and subdomain here. Should simplify the flow creator a bit

**Review by dadolhay — CHANGES_REQUESTED** [review]

File: app/queues/dns/add-record-flow.server.ts (hunk 21 above)
Comment (Genne23v) [reply app/queues/dns/add-record-flow.server.ts:37]: I'm not sure how much I understand the whole system. But I think there would not be many places using `fqdn`.

File: app/queues/dns/add-record-flow.server.ts [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+import { checkDnsStatusQueueName } from './workers/poll-dns-status-worker.server';
+import { syncDbStatusQueueName } from './workers/sync-db-status-worker.server';
+import { createUserRecord } from '~/models/record.server';
+
+import type { FlowJob } from 'bullmq';
+import type { Record } from '@prisma/client';
+import type { DnsUpdaterData } from './workers/dns-update-worker.server';
+import type { DbRecordSynchronizerData } from './workers/sync-db-status-worker.server';
+
+export type SubDomain = { subdomain: string };
```
Comment (humphd) [inline app/queues/dns/add-record-flow.server.ts:14]: `Subdomain` vs. `SubDomain` please.

File: app/queues/dns/update-record-flow.server.ts [hunk 26]
```diff
@@ -0,0 +1,91 @@
...
+import { updateRecordById } from '~/models/record.server';
+
+import type { Record } from '@prisma/client';
+import type { FlowJob } from 'bullmq';
+import type { DnsUpdaterData } from './workers/dns-update-worker.server';
+import type { DbRecordSynchronizerData } from './workers/sync-db-status-worker.server';
+import type { SubDomain } from './add-record-flow.server';
+
+export type UpdateDnsRequestData = Pick<Record, 'id' | 'username' | 'type' | 'value'> &
+  Partial<Pick<Record, 'description' | 'course' | 'ports'>> &
```
Comment (humphd) [inline app/queues/dns/update-record-flow.server.ts:17]: You've become our resident TS master!  I love it.

File: app/queues/dns/workers/poll-dns-status-worker.server.ts [hunk 27]
```diff
@@ -0,0 +1,30 @@
+import { Worker } from 'bullmq';
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import { getChangeStatus } from '~/lib/dns.server';
+import { dnsUpdateQueueName } from './dns-update-worker.server';
+import { getChildrenValuesOfQueueName } from '~/utils';
+
+export const checkDnsStatusQueueName = 'check-dns-status';
+
+export const pollDnsStatusWorker = new Worker(
```
Comment (humphd) [inline app/queues/dns/workers/poll-dns-status-worker.server.ts:10]: How come you don't give this `Worker` a type like all the others?

**Review by humphd — COMMENTED** [review]
Two small things.  This is looking so much better, great work.

File: app/queues/dns/update-record-flow.server.ts (hunk 26 above)
Comment (Genne23v) [reply app/queues/dns/update-record-flow.server.ts:17]: I feel I'm lucky to go through all this!

**Review by humphd — APPROVED** [review]

**Comment by humphd** [comment]
Only took 2 days, not bad!
